multidimensional-pod-autoscaler/AEP.md

```diff
@@ -254,6 +254,11 @@ status:
     value: metric-value
 ```
 
+Note that application SLO field is **optional** and SLO is defined to be the quality of service target that an application must meet (regarding latency, throughput, and so on).
+For example, if the latency SLO is in use, then it could be 99% of the requests finish within 100ms. Accordingly, the replica set can be horizontally scaled when the measured latency is greater than 100ms, i.e., violating the SLO value.
+The default MPA recommender implemented in this AEP will not use the `applicationSLO` field and this field will be used by users who want to implement their own recommender. For example, an RL/ML-based recommender can have `applicationSLO` as part of the reward/loss function and thus they can have extra constraints in addition to min/max replicas.
+The `applicationSLO` field is a floating point number (most application metrics like latency and throughput are floating point numbers).
```

Comment:
It seems to me that just the target number is not enough.

Let's use the example from the doc: we're targeting 99% of requests handled in 100 ms or faster.

It seems that we need to provide at least 2 pieces of information:

1) Metric reporting 99th percentile of request handling time and
2) 100ms (in the same units as used by the metric)

The proposed API has only 1 field of type double, so I can provide 2) (target SLO value) but I don't see a way to specify 1) (metric we're targeting).

For beta I think we ill need ability to specify something more complicated than just a metric value (for example percentile of histogram metric, support for incremental metrics). But for alpha I think we need at least ability to specify what metric we're targetting.

Please amend the proposal to allow that or explain how it's possible with the current proposdal

Reply:
@wangchen615 @James-QiuHaoran 

I still think the proposal doesn't work. Now there are 2 fields:

- `applicationSLO` and
-  `percentageSLO `

This allows us to specify that we want value of 100 or less 99% of the time.

It still doesn't allow us to specify what metric we're targeting. (Which metric should have value 100 or less 99% of the time?)

Maybe it's better to do like @pbetkier suggested and put those parameters in recommender-specific place instead of putting them in the general API?

In addition to what Piotr pointed out (OS recommender is not going to use those fields so it's a bit weird to have them) it looks like this part of the API is not fully ready yet.